src/ws/session-utils.c

```diff
@@ -177,7 +177,7 @@ void
 write_control_string (const char *field,
                       const char *str)
 {
-  json_print_string_property (authf, field, str, -1);
```

Comment:
This change is incorrect.  `-1` means "string is nul terminated" which is exactly the intention here.

Reply:
I'm guessing that covscan isn't enjoying the casual conversion of that signed value one to a signed value later on (and the ensuing comparisons), but that's what I wanted.  It's a max-length field, so a good way to deal with nul is to effective use the maximum value of a size_t as "maximum length".

Is there a way we can annotate this weird behaviour as correct?  Would an explicit cast help?

Reply:
oh right, I did not get that :) I'll remove this change.
I can just waive it in covscan, no problem with that.